dotnet-desktop-guide/net/winforms/visualstudio/troubleshoot-32bit.md

```diff
@@ -0,0 +1,77 @@
+
+Previous versions of Visual Studio targeted 32-bit, and your project probably compiled to `AnyCPU`, which would pick 32-bit while in design mode to match Visual Studio. 32-bit specific references worked, but if you had a 64-bit specific reference, you might have run into a problem with the designer. With Visual Studio 2022, the problem reversed. Visual Studio 2022 is only available in 64-bit. Components and libraries that were compiled as `AnyCPU` work in both 32-bit and 64-bit and don't have an issue running in Visual Studio 2022 64-bit. But, after upgrading to Visual Studio 2022, your projects might fail to run at design-time if the project relies on a 32-bit specific component. This is even the case when your referenced component is compiled for `AnyCPU`, but happens to reference a 32-bit component or 32-bit COM\ActiveX library directly.
+
+To summarize, 32-bit components can't be used by the Windows Forms designer in Visual Studio 2022, which is a 64-bit app. The **out-of-process designer** was created to help Windows Forms for .NET apps during design-time, and is now improved to help with loading 32-bit components.
+
+## What can you do
+
+There are a few design changes you should consider, which might help your project.
+
+- Upgrade from .NET Framework to .NET 8+.
+
+  .NET uses the out-of-process designer, which helps with 32-bit designer problems.
+
+- With .NET Framework, set your app to target `AnyCPU`.
+
+  If you target `AnyCPU` and enable `Prefer 32-bit`, your app runs under 64-bit when in Visual Studio design-time, but compiles to 32-bit for run-time.
+
+- Recompile the 32-bit component for `AnyCPU` or 64-bit.
+
+  If you have access to the source code for the 32-bit component, try compiling it for `AnyCPU` or 64-bit and reference that new version.
+
+- Find a 64-bit alternative component.
+
+  If you're using a component owned by someone else, check to see if they offer a 64-bit version, and reference that.
+
+- Try the out-of-process designer.
+
+  Your final option would be to enable the out-of-process designer for .NET Framework.
+
+## Out-of-process designer
+
+If your project targets .NET 8, you're already using the out-of-process designer. However, if you're still using .NET Framework, you need to enable the out-of-process designer.
```

Comment:
_"If your project targets .NET 8"_

If the project targets any version of .NET starting from .NET Core 3.1 to .NET 9, the out-of-process designer will be used.